webapp/channels/src/components/custom_status/custom_status_emoji.tsx

```diff
@@ -48,6 +45,37 @@ function CustomStatusEmoji({
 
     const customStatusExpired = useSelector((state: GlobalState) => isCustomStatusExpired(state, customStatus));
     const customStatusEnabled = useSelector(isCustomStatusEnabled);
+
+    const [placement, setPlacement] = useState('bottom');
+    const emojiRef = useRef<HTMLSpanElement>(null);
+
+    const handleMouseEnter = useCallback(() => {
+        if (emojiRef.current) {
+            const boundingRect = emojiRef.current.getBoundingClientRect();
+            const windowHeight = window.innerHeight;
+            const threshold = windowHeight * 0.8;
+
+            if (boundingRect.bottom >= threshold) {
+                setPlacement('top');
+            } else {
+                setPlacement('bottom');
+            }
+        }
+    }, [emojiRef]);
```

Comment:
We cannot use a ref as dependency and hence we have to get rid of useCallback for function handleMouseEnter